### PR TITLE
Further constrain `Client m ` in ObeliskWidget

### PR DESCRIPTION
**Pull request by Ericson2314**

### CODE REVIEW

File: lib/frontend/src/Obelisk/Frontend.hs [hunk 1]
```diff
@@ -67,11 +67,14 @@ type ObeliskWidget t route m =
   , MonadFix (Performable m)
   , PrimMonad m
   , Prerender t m
-  , SetRoute t route m
-  , SetRoute t route (Client m)
+  , PrebuildAgonstic t route m
+  , PrebuildAgonstic t route (Client m)
+  )
+
+type PrebuildAgonstic t route m =
```
Comment (alexfmpe) [inline lib/frontend/src/Obelisk/Frontend.hs]: “Agnostic” ?

Comment (Ericson2314) [reply lib/frontend/src/Obelisk/Frontend.hs]: Constraints that apply in the prebuilding and regular building case, and a non-exported name anyways.

Comment (alexfmpe) [reply lib/frontend/src/Obelisk/Frontend.hs]: Yeah that makes sense. I'm just pointing out the typo

**Comment by tomsmalley** [comment]
@Ericson2314 apologies about the conflicts, I just bumped reflex-dom to add the HasJS constraints back in. This makes sense to me, and I didn't even spot the typo :+1:

**Comment by tomsmalley** [comment]
@Ericson2314 thanks!

**Pull request merged**